apps/SageAccountWeb/src/components/ProfileValidation/ReturnToAppButton.tsx

```diff
@@ -11,32 +11,40 @@ import {
   Typography,
   useTheme,
 } from '@mui/material'
-import { cloneElement, PropsWithChildren, ReactElement, useState } from 'react'
+import { ReactNode, useState } from 'react'
 import { NavLink } from 'react-router'
 
-export function ReturnToAppButton({ children }: PropsWithChildren) {
+type ReturnToAppButtonProps = {
+  children?: (onClick?: () => void) => ReactNode
+}
+
+export function ReturnToAppButton({ children }: ReturnToAppButtonProps) {
   const theme = useTheme()
   const [open, setOpen] = useState(false)
   const onClose = () => setOpen(false)
   const redirectFn = () => {
     window.location.assign('/authenticated/myaccount')
   }
-  const element = !children ? (
+  const element = children ? (
+    children(() => setOpen(true))
+  ) : (
```

Comment:
Instead of passing a new prop via `cloneElement`, use the render props pattern and pass a callback that can be consumed by `children`